main/src/ca/uwaterloo/flix/language/phase/Weeder.scala

```diff
@@ -687,23 +687,25 @@ object Weeder {
         case e => WeededAst.Expression.Discard(e, loc)
       }
 
-    case ParsedAst.Expression.ForEach(sp1, pat, exp1, exp2, sp2) =>
+    case ParsedAst.Expression.ForEach(sp1, gens, exp, sp2) =>
       //
       // Rewrites a foreach loop to Foreach.foreach call.
       //
-      val loc = mkSL(sp1, sp2).asSynthetic
-      val patVal = visitPattern(pat)
-      val exp1Val = visitExp(exp1, senv)
-      val exp2Val = visitExp(exp2, senv)
-      val fqn = "ForEach.foreach"
 
-      mapN(patVal, exp1Val, exp2Val) {
-        case (p, e1, e2) =>
-          val lambda = mkLambdaMatch(sp1, p, e2, sp2)
-          mkApplyFqn(fqn, List(lambda, e1), loc)
+      val fqn = "ForEach.foreach"
+      val loc = mkSL(sp1, sp2).asSynthetic
+      val patVals = traverse(gens)(g => visitPattern(g.pat))
```

Comment:
traverse(gens)(visitPattern(_.pat)

Reply:
Doesn't compile with this... for some reason it only accepts `g => visitPattern(g.pat)`

Reply:
Ok